sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -131,181 +164,160 @@ public boolean isEndRequired() {
   }
 
   @Override
-  public void shutdown() {
-    workerThread.interrupt();
-    worker.shutdown();
-  }
+  public CompletableResultCode shutdown() {
+    closed = true;
+    nextExport.cancel(false);
+
+    final CompletableResultCode result = new CompletableResultCode();
+    final CompletableResultCode flushResult = forceFlush();
```

Comment:
It feels like there should be a way to compose the things in here more elegantly, but it could be done as a follow-on PR.

Reply:
Lambdas ;-)